Add envoy connection balancing. (#14616)
hashicorp/consul: website/content/docs/connect/proxies/envoy.mdx

  specified, inherits the Envoy default for route timeouts (15s). A value of 0 will
  disable request timeouts.

- `balance_inbound_connections` - The strategy used for balancing inbound connections

>   across Envoy worker threads. Connect's Envoy integration currently supports the

Reviewer: ```suggestion
  across Envoy worker threads. Consul service mesh Envoy integration supports the
```
I think we're trying to get away from referring to "Connect" as an entity unless we're talking about the configuration `connect`.

  following `balance_inbound_connections` values:

>   - `""` - The default (empty) value. No connection balancing strategy will be used. 
>   - `exact_balance` - If set to this value, inbound connections to the service will
>     use the
>   [Envoy Exact Balance Strategy.](https://cloudnative.to/envoy/api-v3/config/listener/v3/listener.proto.html#config-listener-v3-listener-connectionbalanceconfig-exactbalance)

Reviewer: ```suggestion
  - `""` - Empty string (default). No connection balancing strategy is used. Consul does not balance inbound connections. 
  - `exact_balance` - Inbound connections to the service use the
  [Envoy Exact Balance Strategy.](https://cloudnative.to/envoy/api-v3/config/listener/v3/listener.proto.html#config-listener-v3-listener-connectionbalanceconfig-exactbalance)
```
Use present tense instead of future tense per the style guide. I added a little more information about what happens when the default value is set, but I'm not sure if this is accurate. Please verify.

  -  `enforcing_consecutive_5xx` - The % chance that a host will be actually ejected 
    when an outlier status is detected through consecutive 5xx.

> - `balance_outbound_connections` - The strategy used for balancing outbound connections
>   across Envoy worker threads. Connect's Envoy integration currently supports the
>   following `balance_outbound_connections` values:
> 
>   - `""` - The default (empty) value. No connection balancing strategy will be used. 
>   - `exact_balance` - If set to this value, outbound connections from the upstream will
>     use the
>   [Envoy Exact Balance Strategy.](https://cloudnative.to/envoy/api-v3/config/listener/v3/listener.proto.html#config-listener-v3-listener-connectionbalanceconfig-exactbalance)

Reviewer: ```suggestion
- `balance_outbound_connections` - Specifies the strategy for balancing outbound connections
  across Envoy worker threads. Consul service mesh Envoy integration supports the
  following `balance_outbound_connections` values:

  - `""` - Empty string (default). No connection balancing strategy is used. Consul does not balance outbound connections. 
  - `exact_balance` - Outbound connections from the upstream use the
  [Envoy Exact Balance Strategy.](https://cloudnative.to/envoy/api-v3/config/listener/v3/listener.proto.html#config-listener-v3-listener-connectionbalanceconfig-exactbalance)
```
Same suggestions as inbound param.
